src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp

```diff
@@ -73,25 +90,51 @@ ov::pass::CompressFloatConstantsImpl::CompressFloatConstantsImpl() {
         auto c_type = const_node->get_element_type();
         std::shared_ptr<ov::Node> new_const;
         if (c_type == ov::element::f32) {
-            new_const = change_constant_precision_to_fp16<ov::element::Type_t::f32>(const_node);
+            if (postponed) {  // New optimized implemenation based on count_out_of_f16_range, replace postposed by false
+                              // to disable
+                auto size = shape_size(const_node->get_output_shape(0));
+                auto num_out_of_range =
+                    ngraph::runtime::reference::count_out_of_f16_range(const_node->get_data_ptr<ov::element::f32>(),
+                                                                       size);
+
+                // if more than 75% of a FP32 constant do not fit into FP16 keep in FP32
+                float keep_threshold = 0.75f;
+                float out_of_range_proportion = static_cast<float>(num_out_of_range) / static_cast<float>(size);
+
+                if (out_of_range_proportion >= keep_threshold) {
+                    return false;
+                }
+                new_const = const_node;
+            } else {
+                new_const = change_constant_precision_to_fp16<ov::element::Type_t::f32>(const_node, postponed);
+            }
         } else if (c_type == ov::element::f64) {
-            new_const = change_constant_precision_to_fp16<ov::element::Type_t::f64>(const_node);
+            new_const = change_constant_precision_to_fp16<ov::element::Type_t::f64>(const_node, postponed);
         } else {
             return false;
         }
 
         if (!new_const) {
             return false;
         }
+        auto constant_target_inputs = const_node->get_output_target_inputs(0);
         auto convert = std::make_shared<ov::op::v0::Convert>(new_const, const_node->get_element_type());
 
-        new_const->set_friendly_name(const_node->get_friendly_name() + "_compressed");
         convert->set_friendly_name(const_node->get_friendly_name());
+        new_const->set_friendly_name(const_node->get_friendly_name() + "_compressed");
```

Comment:
Only compression of the constant itself is postponed, graph topology is changed here as well as changes in node attributes are applied here because at this point we have already decided to compress.